app/src/behaviors/behavior_hold_tap.c

```diff
@@ -280,12 +281,85 @@ static inline char *flavor_str(enum flavor flavor) {
         return "balanced";
     case ZMK_BHV_HOLD_TAP_FLAVOR_TAP_PREFERRED:
         return "tap-preferred";
+    default:
+        return "UNKNOWN FLAVOR";
     }
-    return "UNKNOWN FLAVOR";
 }
 
-static void decide_hold_tap(struct active_hold_tap *hold_tap, enum decision_moment event_type) {
-    if (hold_tap->is_decided) {
+static inline char *status_str(enum status status) {
+    switch (status) {
+    case STATUS_UNDECIDED:
+        return "undecided";
+    case STATUS_HOLD_TIMER:
+        return "hold-timer";
+    case STATUS_HOLD_INTERRUPT:
+        return "hold-interrupt";
+    case STATUS_TAP:
+        return "tap";
+    default:
+        return "UNKNOWN STATUS";
+    }
+}
+
+static inline char *decision_moment_str(enum decision_moment decision_moment) {
```

Comment:
Since you are returning constant strings, you *should* return a `const*`. Otherwise you return a pointer to a buffer that someone can try to modify even though it's really const, and then you run into a wonderful thing called undefined behavior.